### PR TITLE
Added DeviceIoControl to Airfs

### PR DESCRIPTION
**Pull request by JohnOberschelp**
Added DeviceIoControl code à la Memfs, simplified file allocation code, and cleaned up some white space.

----

Before submitting this PR please review this checklist. Ideally all checkmarks should be checked upon submitting. (Use an x inside square brackets like so: [x])

- [x] **Contributing**: You MUST read and be willing to accept the [CONTRIBUTOR AGREEMENT](https://github.com/billziss-gh/winfsp/blob/master/Contributors.asciidoc). The agreement gives joint copyright interests in your contributions to you and the original WinFsp author. If you have already accepted the [CONTRIBUTOR AGREEMENT](https://github.com/billziss-gh/winfsp/blob/master/Contributors.asciidoc) you do not need to do so again.
- [ ] **Topic branch**: Avoid creating the PR off the master branch of your fork. Consider creating a topic branch and request a pull from that. This allows you to add commits to the master branch of your fork without affecting this PR.
- [x] **No tabs**: Consistently use SPACES everywhere. NO TABS, unless the file format requires it (e.g. Makefile).
- [x] **Style**: Follow the same code style as the rest of the project.
- [x] **Tests**: Include tests to the extent that it is possible, especially if you add a new feature.
- [x] **Quality**: Your design and code should be of high quality and something that you are proud of.

### CODE REVIEW

File: tst/airfs/airfs.cpp [hunk 1]
```diff
@@ -289,7 +293,7 @@ void DeleteAllNodes(AIRFS_ Airfs)
         else
         {
 #if defined(AIRFS_NAMED_STREAMS)
-			if (Node->Streams)
+            if (Node->Streams)
```
Comment (billziss-gh) [inline tst/airfs/airfs.cpp]: Is this an inadvertent tab? (not sure).

File: tst/airfs/airfs.cpp [hunk 2]
```diff
@@ -334,7 +338,7 @@ NTSTATUS FindNode(AIRFS_ Airfs, PWSTR Name, PWSTR *BaseName,
     {
         //  Isolate the next base name.
         for (fm = to+1; *fm == L'\\'; fm++) {}
-		for (to = fm; *to != L'\0' && *to != L'\\'; to++)
+        for (to = fm; *to != L'\0' && *to != L'\\'; to++)
```
Comment (billziss-gh) [inline tst/airfs/airfs.cpp]: Is this an inadvertent tab? (Not sure.)

File: tst/airfs/airfs.cpp [hunk 3]
```diff
@@ -1419,9 +1417,9 @@ NTSTATUS ApiReadDirectory(FSP_FILE_SYSTEM *FileSystem, PVOID Node0,
     {
         NODE_ Node = *Iter;
         if (!AddDirInfo(Node, Node->Name, Buffer, Length, PBytesTransferred))
-		{
+        {
```
Comment (billziss-gh) [inline tst/airfs/airfs.cpp]: Is this an inadvertent tab? (Not sure.)

File: tst/airfs/airfs.cpp [hunk 4]
```diff
@@ -1419,9 +1417,9 @@ NTSTATUS ApiReadDirectory(FSP_FILE_SYSTEM *FileSystem, PVOID Node0,
...
             goto bufferReady;
-    	}
+        }
```
Comment (billziss-gh) [inline tst/airfs/airfs.cpp]: Is this an inadvertent tab? (Not sure.)

File: tst/airfs/airfs.cpp [hunk 5]
```diff
@@ -1609,6 +1607,32 @@ NTSTATUS ApiGetStreamInfo(FSP_FILE_SYSTEM *FileSystem, PVOID Node0,
...
+            if ((c|0x20) >= 'a' && (c|0x20) <= 'z') c ^= 0x20;
+            ((char*)OutputBuffer)[i] = c;
+        }
+        *PBytesTransferred = InputBufferLength;
+        return STATUS_SUCCESS;
+    }
+
+    return STATUS_INVALID_DEVICE_REQUEST;
+}
+
```
Comment (billziss-gh) [inline tst/airfs/airfs.cpp]: Thanks for this.

I am wondering though if we should use rot13 to keep it inline with MEMFS. Winfsp-tests currently checks for rot13 in its `DeviceIoControl` test:

https://github.com/billziss-gh/winfsp/blob/master/tst/winfsp-tests/devctl-test.c#L52

**Comment by billziss-gh** [comment]
John, thanks for the PR. Some comments inline.

File: tst/airfs/airfs.cpp (hunk 5 above)
Comment (JohnOberschelp) [reply tst/airfs/airfs.cpp]: Of course! And those indent changes are (supposed to be at least) replacing a few tabs that snuck into my code with spaces.

Comment (billziss-gh) [reply tst/airfs/airfs.cpp]: Thanks!

**Comment by billziss-gh** [comment]
John, thank you for the excellent work as usual. Your PR is merged.

**Pull request merged**